server/services/survey.service.js

```diff
@@ -0,0 +1,60 @@
+const Path = require('path-parser');
+const { URL } = require('url');
+
+module.exports = class SurveyService {
+
+  async updateSurveys(data) {
+    try {
+
+      const uniqueServeys =
+          data.map(this.buildSurveyDataFromBody)
+              .filter(this.filterOutNullAndUndefined)
+              .filter(this.filterByUniqueEmailAndSurveyId);
+
+      for (let survey of uniqueServeys) {
+        await Survey.updateOne(
+            {
+              _id: survey.surveyId,
+              recipients: {
+                $elemMatch: {email: survey.email, responded: false}
+              }
+            },
+            {
+              $inc: {[survey.choice]: 1},
+              $set: {'recipients.$.responded': true},
+              lastResponded: new Date()
+            }
+        ).exec();
+      }
+
+      return { success: true }
+    } catch(error) {
+      return error;
+    }
+  }
+
+  getRecipientListFromEmailString(recipients) {
+    return recipients.split(',').map(email => {
+      return { email };
+    })
+  }
+
+
+  buildSurveyDataFromBody(bodyItem) {
+    const path = new Path('/api/surveys/:surveyId/:choice');
+    const match = path.test(new URL(surveyData.url).pathname);
+    if (match) {
+      return { email: surveyData.email, surveyId: match.surveyId, choice: match.choice };
+    }
+  }
+
+  filterOutNullAndUndefined(item) {
+    return item !== null && item !== undefined;
+  }
+
+  filterByUniqueEmailAndSurveyId(survey, index, array) {
+    const emailSurveyIdKeyArray = array.map(item => item.email + item.surveyId);
+
+    return emailSurveyIdKeyArray.indexOf(survey.email + survey.id) === index;
+  }
```

Comment:
This part replaces lodash, it's another filter method here. The lodash method was actually broken, as he tried to make it unique by both email and survey ID, but if you check the lodash docuemtation, uniqBy only takes a single string to be unique by. So...his old code was only unique by email, not survey ID. Meaning...it was broken.

This works properly and filters down to a unique list by email and survey ID, by creating a new array with a key string of email + survey ID, and testing if the index of the current email + id is the same as the current index.